pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraContributors.scala

```diff
@@ -162,4 +162,45 @@ object SierraContributors
       case o: Organisation[IdState.Unminted] => o.copy(id = id)
       case m: Meeting[IdState.Unminted]      => m.copy(id = id)
     }
+
+  /* Given an agent and the associated MARC subfields, look for instances of subfield $0,
+ * which are used for identifiers.
+ *
+ * This methods them (if present) and wraps the agent in Unidentifiable or Identifiable
+ * as appropriate.
+ */
+  private def identify(subfields: List[Subfield],
+                       ontologyType: String,
+                       label: String): IdState.Unminted = {
+
+    // We take the contents of subfield $0.  They may contain inconsistent
+    // spacing and punctuation, such as:
+    //
+    //    " nr 82270463"
+    //    "nr 82270463"
+    //    "nr 82270463.,"
+    //
+    // which all refer to the same identifier.
+    //
+    // For consistency, we remove all whitespace and some punctuation
+    // before continuing.
+    val codes = subfields.collect {
+      case Subfield("0", content) => content.replaceAll("[.,\\s]", "")
+    }
+
+    // If we get exactly one value, we can use it to identify the record.
+    // Some records have multiple instances of subfield $0 (it's a repeatable
+    // field in the MARC spec).
+    codes.distinct match {
+      case Seq(code) =>
+        IdState.Identifiable(
+          SourceIdentifier(
+            identifierType = IdentifierType.LCNames,
+            value = code,
+            ontologyType = ontologyType
+          )
+        )
+      case _ => addIdentifierFromText(ontologyType, label)
```

Comment:
This is the one change from the previous method – we'll now add a label-derived identifier to contributors rather than leaving them unidentified.